Godeps/_workspace/src/github.com/docker/distribution/registry/storage/layerwriter.go

```diff
@@ -298,8 +336,13 @@ func (lw *layerWriter) validateLayer(dgst digest.Digest) (digest.Digest, error)
 	)
 
 	if lw.resumableDigester != nil {
+		offset := lw.size
+		if offset < lw.offset {
+			// we're probably on NFS where file attributes are cached on client
+			offset = lw.offset
```

Comment:
Yeah, I was also concerned by that, but looking more I'm thinking its using the offset you set after preforming previous operation. So that's your internal state, iiuc.